src/main/java/network/brightspots/rcv/RawContestConfig.java

```diff
@@ -96,162 +100,233 @@ public static class CvrSource {
+      this.idColumnIndex.set(idColumnIndex);
+      this.precinctColumnIndex.set(precinctColumnIndex);
+      this.overvoteDelimiter.set(overvoteDelimiter);
+      this.provider.set(provider);
+      this.contestId.set(contestId);
+      this.overvoteLabel.set(overvoteLabel);
+      this.undervoteLabel.set(undervoteLabel);
+      this.undeclaredWriteInLabel.set(undeclaredWriteInLabel);
+      this.treatBlankAsUndeclaredWriteIn.set(treatBlankAsUndeclaredWriteIn);
     }
 
     public String getFilePath() {
-      return filePath;
+      return filePath.get();
     }
 
     public void setFilePath(String filePath) {
-      this.filePath = filePath;
+      this.filePath.set(filePath);
     }
 
     public String getContestId() {
-      return contestId;
+      return contestId.get();
     }
 
     public void setContestId(String contestId) {
-      this.contestId = contestId;
+      this.contestId.set(contestId);
     }
 
     public String getFirstVoteColumnIndex() {
-      return firstVoteColumnIndex;
+      return firstVoteColumnIndex.get();
     }
 
     public void setFirstVoteColumnIndex(String firstVoteColumnIndex) {
-      this.firstVoteColumnIndex = firstVoteColumnIndex;
+      this.firstVoteColumnIndex.set(firstVoteColumnIndex);
     }
 
     public String getFirstVoteRowIndex() {
-      return firstVoteRowIndex;
+      return firstVoteRowIndex.get();
     }
 
     public void setFirstVoteRowIndex(String firstVoteRowIndex) {
-      this.firstVoteRowIndex = firstVoteRowIndex;
+      this.firstVoteRowIndex.set(firstVoteRowIndex);
     }
 
     public String getIdColumnIndex() {
-      return idColumnIndex;
+      return idColumnIndex.get();
     }
 
     public void setIdColumnIndex(String idColumnIndex) {
-      this.idColumnIndex = idColumnIndex;
+      this.idColumnIndex.set(idColumnIndex);
     }
 
     public String getPrecinctColumnIndex() {
-      return precinctColumnIndex;
+      return precinctColumnIndex.get();
     }
 
     public void setPrecinctColumnIndex(String precinctColumnIndex) {
-      this.precinctColumnIndex = precinctColumnIndex;
+      this.precinctColumnIndex.set(precinctColumnIndex);
     }
 
     public String getOvervoteDelimiter() {
-      return overvoteDelimiter;
+      return overvoteDelimiter.get();
     }
 
     public void setOvervoteDelimiter(String overvoteDelimiter) {
-      this.overvoteDelimiter = overvoteDelimiter;
+      this.overvoteDelimiter.set(overvoteDelimiter);
     }
 
     public String getProvider() {
-      return provider;
+      return provider.get();
     }
 
-    public void setProvider(String provider) {
-      this.provider = provider;
+    /**
+     * Set the provider by either its GUI label or its internal label.
+     * Supporting both means we don't have to translate between internal and GUI labels
+     * each time we serialize/deserialize in the table.
+     *
+     * @param providerString The provider's internal or GUI label.
+     */
```

Comment:
I noticed that in the GUI table, it now shows the non-user-friendly versions, like "ess" and "clearBallot" instead of "ES&S" and "Clear Ballot". Is that an artifact of this? Can we revert to that more user-friendly behavior?

Reply:
done